index.js

```diff
@@ -4,14 +4,49 @@ module.exports = robot => {
   robot.on(['pull_request.opened', 'pull_request.synchronize'], check)
 
   async function check (context) {
+    const config = await context.config('dco.yml', {'require_signoff_for_members': true})
+    const requireSignoffForMembers = config.require_signoff_for_members
+
     const pr = context.payload.pull_request
+    const organization = context.payload.organization.login
 
     const compare = await context.github.repos.compareCommits(context.repo({
       base: pr.base.sha,
       head: pr.head.sha
     }))
 
-    const dcoParams = getDCOStatus(compare.data.commits)
+    const members = {}
+
+    const isOrgMember = async function (login) {
+      if (members.hasOwnProperty(login)) {
+        return members[login]
+      }
+
+      let result
+      try {
+        await context.github.orgs.checkMembership({
+          org: organization,
+          username: login
+        })
+        result = true
+      } catch (err) {
+        if (err.code === 404) {
+          result = false
+        } else {
+          throw err
+        }
+      }
```

Comment:
I love `async`/`await`, but I've found it can be very verbose in situations like this. I've used this pattern a few times recently, but still don't love it:

```js
const result = await context.github.orgs.checkMembership({
  org: organization,
  username: login
}).catch(err => {
  if (err.code !== 404) {
    throw err
  }
  return false
})
```

Reply:
Thanks. I'm new to this async JavaScript game, and this version indeed looks somewhat better. Changed.